samples/durable-task-sdks/dotnet/Agents/PromptChaining/Worker/Services/IAgentService.cs

```diff
@@ -0,0 +1,440 @@
+using System.Text.Json;
+using Azure;
+using Azure.AI.Projects;
+using Azure.AI.Agents.Persistent;
+using Azure.Core;
+using Azure.Identity;
+using Microsoft.Extensions.Logging;
+using System.Text;
+
+namespace AgentChainingSample.Services;
+
+/// <summary>
+/// Interface for agent services
+/// </summary>
+public interface IAgentService
+{
+    /// <summary>
+    /// Gets the agent ID used by this service
+    /// </summary>
+    string AgentId { get; set; }
+
+    /// <summary>
+    /// Gets the endpoint URI for this agent service
+    /// </summary>
+    string Endpoint { get; }
+    
+    /// <summary>
+    /// Ensures the agent exists, creating it if necessary
+    /// </summary>
+    Task<string> EnsureAgentExistsAsync(string agentName, string systemPrompt);
+
+    /// <summary>
+    /// Gets a response from the agent
+    /// </summary>
+    Task<string> GetResponseAsync(string prompt);
+
+    /// <summary>
+    /// Cleans JSON response from markdown formatting
+    /// </summary>
+    string CleanJsonResponse(string response);
+}
+
+/// <summary>
+/// Base implementation for agent services
+/// </summary>
+public abstract class BaseAgentService : IAgentService
```

Comment:
Nit: I'd move the implementation types to their own files so they're easier to locate.